app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/movielist/view/adapter/MovieListAdapter.kt

```diff
@@ -17,16 +19,20 @@ class MovieListAdapter : RecyclerView.Adapter<MovieListAdapter.BindingHolder>()
 
     override fun onBindViewHolder(holder: BindingHolder, position: Int) {
         val data = movies[position]
-        holder.binding.singleItemViewModel = SingleItemViewModel(data)
+        holder.binding.singleMovieViewModel = SingleMovieViewModel(data)
     }
 
     override fun getItemCount(): Int {
         return movies.size
     }
 
-    fun setMovies(data: List<Movie>) {
-        movies.clear()
-        movies.addAll(data)
+    fun setMovies(data: List<Movie>?) {
+        movies.run {
+            clear()
+            val addAll = data?.let { this.addAll<Movie>(it) }
```

Comment:
What is the reason of adding val addAll ?
Can it be simple:
addAll(it) 
?

Reply:
this problem is connected with the custom binding adapter implementation connected with the problem with avoiding observing list in fragment - i will try to fix it in the second part of this task